rundeckapp/grails-app/controllers/rundeck/controllers/ProjectController.groovy

```diff
@@ -468,6 +473,24 @@ class ProjectController extends ControllerBase{
         }
     }
 
+    /**
+     * Give a hint based on the error message in the exception thrown during the import
+     *
+     * @param Throwable - The exception thrown
+     * @return String - null if is not supported yet, a message if it is.
+     *
+     * */
+    @CompileStatic
+    private String hintErrorCause(Throwable t){
+        final SIZE_CONSTRAINT_VIOLATION_STRING = 'Data too long for column \'data\''
+        final SIZE_CONSTRAINT_VIOLATION_MESSAGE = "The content of the node resource definitions could be too big to store it on the database, try to split it into multiple files before importing it."
```

Comment:
I think it is too specific.  The same exception could be caused if the resource is too large, e.g. project.properties, not necessarily the node sources.  Perhaps saying something like "Some content was too large, this may be caused by a node source definition, ... etc"

Reply:
Agree :+1: